src/app/tests/TestReportingEngine.cpp

```diff
@@ -176,6 +225,100 @@ void TestReportingEngine::TestMergeOverlappedAttributePath(nlTestSuite * apSuite
     InteractionModelEngine::GetInstance()->GetReportingEngine().Shutdown();
 }
 
+bool TestReportingEngine::InsertToDirtySet(const AttributePathParams & aPath)
+{
+    auto path = InteractionModelEngine::GetInstance()->GetReportingEngine().mGlobalDirtySet.CreateObject();
+    VerifyOrReturnError(path != nullptr, false);
+    *path             = aPath;
+    path->mGeneration = InteractionModelEngine::GetInstance()->GetReportingEngine().GetDirtySetGeneration();
+    return true;
+}
+
+void TestReportingEngine::TestMergeAttributePathWhenDirtySetPoolExhausted(nlTestSuite * apSuite, void * apContext)
+{
+    TestContext & ctx = *static_cast<TestContext *>(apContext);
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+    err               = InteractionModelEngine::GetInstance()->Init(&ctx.GetExchangeManager());
```

Comment:
I believe this function requires another argument. It doesn't build for me after syncing this commit.

Reply:
Same  here.  Posted build errors below.